src/unix/signal.c

```diff
@@ -234,20 +252,25 @@ static int uv__signal_register_handler(int signum, int oneshot) {
   if (oneshot)
     sa.sa_flags |= SA_RESETHAND;
 
-  /* XXX save old action so we can restore it later on? */
-  if (sigaction(signum, &sa, NULL))
+  if (sigaction(signum, &sa, &sa_old))
     return UV__ERR(errno);
 
+  uv__sigaction_set(signum, &sa_old);
+
   return 0;
 }
 
 
 static void uv__signal_unregister_handler(int signum) {
   /* When this function is called, the signal lock must be held. */
   struct sigaction sa;
-
-  memset(&sa, 0, sizeof(sa));
-  sa.sa_handler = SIG_DFL;
+  
+  if (uv__sigaction_isset(signum)) {
```

Comment:
Do you mean to just assert this, as it should not be possible to call `uv__signal_unregister_handler` unless `uv__signal_register_handler` was called previously to register that signal handler.
```suggestion
  assert(uv__sigaction_isset(signum));
```

Reply:
@vtjnash Thanks, updated.